x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/get_dataset_info/index.ts

```diff
@@ -32,68 +34,87 @@ export function registerGetDatasetInfoFunction({
           index: {
             type: 'string',
             description:
-              'index pattern the user is interested in or empty string to get information about all available indices',
+              'Index pattern the user is interested in or empty string to get information about all available indices. You are allowed to use wildcards like `logs*`.',
           },
         },
         required: ['index'],
       } as const,
     },
-    async ({ arguments: { index }, messages, chat }, signal) => {
-      const coreContext = await resources.context.core;
+    async ({ arguments: { index: indexPattern }, messages, chat }, signal) => {
+      return getDatasetInfo({ resources, indexPattern, signal, messages, chat });
+    }
+  );
+}
 
-      const esClient = coreContext.elasticsearch.client;
-      const savedObjectsClient = coreContext.savedObjects.client;
+export async function getDatasetInfo({
+  resources,
+  indexPattern,
+  signal,
+  messages,
+  chat,
+}: {
+  resources: RespondFunctionResources;
+  indexPattern: string;
+  signal: AbortSignal;
+  messages: Message[];
+  chat: FunctionCallChatFunction;
+}) {
+  const coreContext = await resources.context.core;
 
-      let indices: string[] = [];
+  const esClient = coreContext.elasticsearch.client;
+  const savedObjectsClient = coreContext.savedObjects.client;
 
-      try {
-        const body = await esClient.asCurrentUser.indices.resolveIndex({
-          name: index === '' ? ['*', '*:*'] : index.split(','),
-          expand_wildcards: 'open',
-        });
-        indices = [
-          ...body.indices.map((i) => i.name),
-          ...body.data_streams.map((d) => d.name),
-          ...body.aliases.map((d) => d.name),
-        ];
-      } catch (e) {
-        indices = [];
-      }
+  let indices: string[] = [];
 
-      if (index === '') {
-        return {
-          content: {
-            indices,
-            fields: [],
-          },
-        };
-      }
+  try {
+    const name = indexPattern === '' ? ['*', '*:*'] : `${indexPattern.split(',')}*`;
```

Comment:
@dgieselaar As discussed I've added a trailing wildcard. I think this is better although this still doesn't find zookeeper logs if the user asks "please show me my zookeeper logs" and they use datastreams with default naming, eg. indices will be something like `.ds-logs-zookeeper.1-default-2025.03.04-000001` and the data stream will be `logs-zookeeper.1-default`.

Reply:
To handle that we'd need both a leading and trailing wildcard (`*${indexPattern.split(',')}*`). I wouldn't mind that but you had some concerns?

Reply:
Maybe we can:
- first execute with the raw value for `index` (let's say `logs`)
- if no indices are resolved, execute it with `logs*`
- if still no indices are resolved, execute it with `*logs*`

we can also always add a `*:...` additionally.

_resolve/index should be relatively fast so I'm not worried about performance.

Reply:
Apart from code complexity it could also be confusing to users how it sometimes includes some indices depending on what other indices they have in their system, eg. if they have a datastream called `zookeeper-legacy-metrics` and another `logs-zookeeper.1-default` it will only include their logs datastream if they delete the legacy datastream.

Reply:
... I think there plenty layers to debug already so whenever we can remove a code branch the better imo.

Reply:
Added an API test for this: https://github.com/elastic/kibana/pull/213231/files#diff-2791f845c2c4184ca75a917c23696cd96e299e9d44aeaf16a717ddc2e91d9c52R291-R321

Reply:
That way we do not match backing indices.